web-app/src/games/api.admin.games.service.ts

```diff
@@ -14,7 +40,74 @@ export class ApiAdminGamesService {
     return this.gamesRepository.findOne(id);
   }
 
-  findMany(): Promise<Game[]> {
-    return this.gamesRepository.find();
+  findMany(searchParams?: GamesSearchParams): Promise<Game[]> {
+    const findOptions: FindManyOptions<Game> = {};
+    if (searchParams) {
+      if (searchParams.sort) {
+        findOptions.order = this.convertToFindOptionsOrder(searchParams.sort);
+      }
+      if (searchParams.filter) {
+        findOptions.where = this.convertToFindOptionsWhere(searchParams.filter);
+      }
+      if (searchParams.skip) findOptions.skip = searchParams.skip;
+      if (searchParams.take) findOptions.take = searchParams.take;
+    }
+    return this.gamesRepository.find(findOptions);
+  }
+
+  deleteOne(id: number) {
+    return this.gamesRepository.delete(id);
+  }
+
+  deleteMany(ids: number[]) {
+    if (!ids) {
+      throw Error('A list of ids must be provided to delete games');
+    }
+    return this.gamesRepository.delete(ids);
+  }
+
+  convertToFindOptionsOrder(sort: GamesSortColumn[]): GameOrderType {
+    const order: GameOrderType = {};
+    sort.forEach((s) => {
```

Comment:
We can use plural here for the sort object, and give a better name for the var inside the loop !